src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+	protected boolean isPrivate;
+	public String value;
+	private static String MESSAGE_CONSTRAINTS;
+	private static String VALIDATION_REGEX;
+
+	public Contact(String value, boolean isPrivate, String messageContraints, String validationRegex)
+			throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		this.value = value;
+		MESSAGE_CONSTRAINTS = messageContraints;
+		VALIDATION_REGEX = validationRegex;
+		this.value = value.trim();
+		if (!isValid(value)) {
+			throw new IllegalValueException(MESSAGE_CONSTRAINTS);
+		}
+	}
+
+	public boolean isPrivate() {
```

Comment:
Use naming conventions for common methods. Therefore, this method should be called `getIsPrivate`. Eclipse can automatically generate the getter or/and setter function for you.